AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -54,6 +56,13 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
         self.mutationEventPublisher = PassthroughSubject<ReconcileAndLocalSaveOperationEvent, DataStoreError>()
 
         self.cancellables = Set<AnyCancellable>()
+        
+        // `isEagerLoad` is true by default, unless the models contain the rootPath
+        // which is indication that codegenerated model types support for lazy loading.
+        if isEagerLoad && ModelRegistry.modelType(from: modelSchema.name)?.rootPath != nil {
+            self.isEagerLoad = false
+        }
```

Comment:
We need to control the eagerload or lazyloading, used by the queries that result in models encoded in MutationEvents. If the Model types do not support lazy loading (LazyReference encapsulation) then we need to continue eager loading on queries, much like the `DataStore.query` API path.